### DISCUSSION TITLE
Not sure which datalogger/device to ask for

### DISCUSSION THREAD

**Discussion by mdeakin99** [alienatedsec/solis-ha-modbus-cloud#31] [Q&A]
Hi

I am getting a Solis AC Coupled inverter, the S6-EA1P(3.6-6)K-L I suspect and I want to make sure I ask for the right data logger.  I am looking to make use of your integration for the following

1) Read vales off of the inverter including power and battery state
2) If possible,  switch to difference modes e.g. force charge/discharge etc

I have looked into ModBus before but can never quite figure our what hardware I need and what wiring I might need to do in order to get it working.

Would you be able to recommend the following so I can do solis cloud,  local and remote control if possible?

1) Which datalogger should I be asking for?
2) Do I need to also buy another modbus piece of hardware e.g. the waveshare devices and if so which one?

I can have a power socket where the inverter is but I wont have LAN connectivity there,  only wifi.

Thanks for all your patience and help!

**Comment by alienatedsec** [comment]
Hi @mdeakin99 

Unfortunately, this project is aiming for inverters with a 4-pin connector datalogger. Your inverter needs a USB datalogger, which I had no chance to investigate further.

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/903a7635-0af4-4814-9316-bcf1d1ebf894)

1. There are plenty of 4-pin dataloggers, which allow certain things, but we know very little about USB dataloggers.
2. Having additional hardware is highly likely required; however, I would not know which hardware could work the same way (or at least allow the same hybrid functionality) as this project.

**Comment by mdeakin99** [comment]
Thanks for taking the time to answer.  I will see what device gets installed in the end and check back here if it turns out to have a four pin connector.

Mark

On 16 Feb 2024, at 22:00, alienatedsec ***@***.***> wrote:

﻿

Hi @mdeakin99<https://github.com/mdeakin99>

Unfortunately, this project is aiming for inverters with a 4-pin connector datalogger. Your inverter needs a USB datalogger, which I had no chance to investigate further.

image.png (view on web)<https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/903a7635-0af4-4814-9316-bcf1d1ebf894>

  1.  There are plenty of 4-pin dataloggers, which allow certain things, but we know very little about USB dataloggers.
  2.  Having additional hardware is highly likely required; however, I would not know which hardware could work the same way (or at least allow the same hybrid functionality) as this project.
